service/src/main/java/org/sunbird/service/user/UserProfileReadService.java

```diff
@@ -133,6 +133,7 @@ public Response getUserProfileData(Request actorMessage) {
     appendMinorFlag(result);
     // For Backward compatibility , In ES we were sending identifier field
     result.put(JsonKey.IDENTIFIER, userId);
+    mapUserRoles(result);
```

Comment:
This should not be done as the SB v5 APIs updates roles value with list of objects instead of list of string.

Reply:
Removed mapUserRoles as it over writes the latest implementation. Will use the latest API in iGOT